### ISSUE TITLE
How to merge documentation from several MPP modules

### ISSUE THREAD

**Issue by sandwwraith** [Kotlin/dokka#776]
**Question**
I have in root build.gradle:
```gradle
apply plugin: 'kotlin'
apply plugin: 'org.jetbrains.dokka'

dokka {
    outputFormat = 'html'
    outputDirectory = "$buildDir/dokka"

    subProjects = ["kotlinx-serialization-runtime", "kotlinx-serialization-protobuf"]
}
```

and in corresponding submodules:

```gradle
apply plugin: 'org.jetbrains.dokka'

dokka {
    multiplatform {
        common {}
        jvm {}
        js {}
        'native' {}
    }
}
```

However, when I run `./gradlew dokka`, I got html with no classes in `build/dokka`. Documentation in modules (`kotlinx-serialization-runtime/build/dokka`) is generated correctly. Also I get a lot of 'Can't find node by signature `kotlinx.serialization.Serializable`, referenced at kotlinx.serialization.protobuf.ProtoBuf (ProtoBuf.kt:17). This is probably caused by invalid configuration of cross-module dependencies'. How do I configure these ones?

**Installation**
- Operating system: macOS 10.15
- Build tool: Gradle v5.6.1
- Dokka version: 0.10.1

**Comment by kamilok1965** [comment]
You should also include the platforms to document in the root dokka config, so 
```groovy
dokka {
    outputFormat = 'html'
    outputDirectory = "$buildDir/dokka"

    subProjects = ["kotlinx-serialization-runtime", "kotlinx-serialization-protobuf"]
}
```
becomes 
```groovy
dokka {
    outputFormat = 'html'
    outputDirectory = "$buildDir/dokka"

    subProjects = ["kotlinx-serialization-runtime", "kotlinx-serialization-protobuf"]

    multiplatform {
        common {}
        jvm {}
        js {}
        'native' {}
    }
}
```

**Comment by kamilok1965** [comment]
Can't find node `kotlinx.serialization.Serializable` can be caused by 3 things essentially: 
1) `kotlinx.serialization.Serializable` is in some other module, you have to configure the passes with external documentation links
2) `kotlinx.serialization.Serializable` is a dependency that's not in the pass classpath (you can run dokka with `--info` to check the classpath for each module)
3) `kotlinx.serialization.Serializable` is in some unpacked klib (which I guess is most likely). This itself is a bug which has a workaround on `native-stdlib-fix` and `native-stdlib-fix-1.3.70` IIRC. Please note that due to klibs having an unstable API they can only be analyzed by the same compiler version used to create them. Dokka have a compiler in version 1.3.61 and there is a dev version with 1.3.70 (the one from `native-stdlib-fix-1.3.70` branch). If that doesn't fix your problem I can only advise right now to analyse the native platform like `common`:
  ```
'native' {
    platform = "common
}
```
and provide a jar with `kotlinx.serialization.Serializable` and other dependencies instead of klibs. Those problems will be fixed when klibs have a stable API

**Comment by sandwwraith** [comment]
I've used your advice and the problem with 'can't find node' is gone too, thanks. 

I now have question about merged documentation: `"kotlinx-serialization-runtime"` and  `"kotlinx-serialization-protobuf"` are different submodules that are published as different maven artifacts.  Merged documentation is generated as they're the one module (no 'modules' section, only 'packages' with packages from both modules). I want them to be different modules, like in kotlinx.coroutines (https://kotlin.github.io/kotlinx.coroutines/): each module has its own packages list. Is such functionality supported out-of-box with `html` output format? Coroutines have quite compilcated setup with markdown and Jekyll I'm trying to avoid. I've tried to apply dokka plugin for each subproject separately and then copy all files into one folder, but I don't know how to generate index.html with modules. Thanks in advance!
